src/pyaro/csvreader/CSVTimeseriesReader.py

```diff
@@ -4,6 +4,25 @@
 import pyaro.timeseries.AutoFilterReaderEngine
 
 
+class ExtededStation(Station):
+    def __init__(self, fields: dict = None) -> None:
+        super().__init__(fields)
+
+    def set_fields(self, fields: dict):
+        """
+
+        :param fields: dict with the required fields
+
+        """
+        if abs(fields["latitude"]) > 90:
+            raise Exception(f"latitude out of bounds: {fields['latitude']}")
+        if abs(fields["longitude"]) > 180:
+            raise Exception(f"longitude out of bounds: {fields['longitude']}")
+        for key in fields:
+            self._fields[key] = fields[key]
+        return
+
+
```

Comment:
It's always ugly to extend a class which is not thought to be extended. Why not add a unspecified "metadata"={} field to the Station-class?